src/openms/include/OpenMS/FORMAT/TriqlerFile.h

```diff
@@ -0,0 +1,222 @@
+                      const String& condition);
+        
+    private:
+
+      typedef OpenMS::Peak2D::IntensityType Intensity;
+      typedef OpenMS::Peak2D::CoordinateType Coordinate;
+
+      static const String na_string_;
+      static const char delim_ = ',';
+      static const char accdelim_ = ';';
+      static const char quote_ = '"';
+
+      /*
+        *  @brief: Struct to aggregate intermediate information from ConsensusFeature and ConsensusMap,
+        *  such as filenames, intensities, retention times, labels and features (for further processing)
+        */
+      struct AggregatedConsensusInfo
+      {
+        std::vector< std::vector< String > > consensus_feature_filenames;           //< Filenames of ConsensusFeature
+        std::vector< std::vector< Intensity > > consensus_feature_intensities;       //< Intensities of ConsensusFeature
+        std::vector< std::vector< Coordinate > > consensus_feature_retention_times; //< Retention times of ConsensusFeature
+        std::vector< std::vector< unsigned > > consensus_feature_labels;          //< Labels of ConsensusFeature
+        std::vector<BaseFeature> features;                                        //<s Features of ConsensusMap
+      };
+
+      /*
+        *  @brief: Aggregates information from ConsensusFeature and ConsensusMap,
+        *  such as filenames, intensities, retention times, labels and features.
+        *  Stores them in AggregatedConsensusInfo for later processing
+        */
+      TriqlerFile::AggregatedConsensusInfo aggregateInfo_(const ConsensusMap& consensus_map,
+                                                          const std::vector<String>& spectra_paths);
+
+      /*
+        *  @brief: Internal function to check if condition exists in Experimental Design
+        */
+      static void checkConditionLFQ_(const ExperimentalDesign::SampleSection& sampleSection, const String& condition);
+
+      /*
+        *  In OpenMS, a run is split into multiple fractions.
+        */
+      static void assembleRunMap_(
+              std::map< std::pair< String, unsigned>, unsigned> &run_map,
+              const ExperimentalDesign &design);
+
+      /*
+        * @brief checks two vectors for same content
+        */
+      static bool checkUnorderedContent_(const std::vector< String> &first, const std::vector< String > &second);
+
+      OpenMS::Peak2D::IntensityType sumIntensity_(const std::set< OpenMS::Peak2D::IntensityType > &intensities) const
+      {
+        OpenMS::Peak2D::IntensityType result = 0;
+        for (const OpenMS::Peak2D::IntensityType &intensity : intensities)
+        {
+          result += intensity;
+        }
+        return result;
+      }
+
+      OpenMS::Peak2D::IntensityType meanIntensity_(const std::set< OpenMS::Peak2D::IntensityType > &intensities) const
+      {
+        return sumIntensity_(intensities) / intensities.size();
+      }
+
+      class TriqlerLine_
```

Comment:
this class could be completely hidden in the implementation. Its not used externally, right?
Or would there be an advantage to let the world know that this exists?